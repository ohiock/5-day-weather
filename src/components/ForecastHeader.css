.day-header {
  margin: 0 0 1vw 0;

  font-size: 3vw;
}
