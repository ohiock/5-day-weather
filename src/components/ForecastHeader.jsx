import React from 'react'
import PropTypes from 'prop-types';

const propTypes = {
  weekDay: PropTypes.string.isRequired,
  today: PropTypes.string.isRequired,
  tomorrow: PropTypes.string.isRequired,
};

const ForecastHeader = (props) => {
  switch (props.weekDay) {
    case props.today:
      return (<h2>Today</h2>);
    case props.tomorrow:
      return (<h2>Tomorrow</h2>);
    default:
      return (<h2>{props.weekDay}</h2>);
  }
}

ForecastHeader.propTypes = propTypes;

export default ForecastHeader;
