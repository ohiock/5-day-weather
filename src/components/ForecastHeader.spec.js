import React from 'react';
import renderer from 'react-test-renderer';
import moment from 'moment';

import ForecastHeader from './ForecastHeader';

it('renders correctly', () => {
  const tree = renderer.create(
    <ForecastHeader
      weekDay={moment().format('dddddd')}
      today={moment().format('dddd')}
      tomorrow={moment().add(1, 'days').format('dddd')} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
