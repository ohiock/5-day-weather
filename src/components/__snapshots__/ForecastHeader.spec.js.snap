// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className={undefined}
>
  <h2
    className={undefined}
  >
    ThursdayTh
  </h2>
</div>
`;
